src/components/appointments/CreateAppointment/CreateAppointment.stories.js

```diff
@@ -131,7 +131,7 @@ export const HappyFlow = {
     await step('Fill out the location and time', async () => {
       expect(await canvas.findByText('Locatie en tijd')).toBeVisible();
       await waitFor(async () => {
-        await expect(canvas.getByRole('button', {name: 'Terug naar producten'})).toBeVisible();
+        await expect(canvas.getByText('Terug naar producten')).toBeVisible();
```

Comment:
I'd use `getByRole('link', {name: ...})` to lock in the accessible behavior via tests